app/views/play/cinematic/Command/CinematicParser.js

```diff
@@ -13,33 +45,57 @@ export const parseShot = (shot, systems) => {
  * Returns an array of Commands for the setup of the shot.
  * @param {Shot} shot
  * @param {Systems} systems
+ * @returns {AbstractCommand[]} The commands to run at the start and end of the shot.
+ */
+const parseSetup = (shot, systems) =>
+  Object.values(systems)
+    .filter(sys => sys !== undefined && typeof sys.parseSetupShot === 'function')
+    .reduce((commands, sys) => {
+      const systemCommands = sys.parseSetupShot(shot)
+      if (!Array.isArray(systemCommands)) {
+        throw new Error('Your system should always return an array of commands.')
+      }
+      return [...commands, ...systemCommands]
+    }
+    , [])
+
+/**
+ * Returns an array of commands.
+ * @param {*} dialogNode 
+ * @param {*} systems 
  */
-const parseSetup = (shot, systems) => {
-  const { cinematicLankBoss, loader } = systems
-  const setupCommands = []
+const parseDialogNode = (dialogNode, systems) =>
+  Object.values(systems)
+    .filter(sys => sys !== undefined && typeof sys.parseDialogNode === 'function')
+    .reduce((commands, sys) => (
+      [...commands, ...sys.parseDialogNode(dialogNode)]
```

Comment:
Do we not need to check if `sys.parseDialogNode(dialogNode)` returns an array similar to `parseSetup` above?

Reply:
Great comment. I had apparently not done this in this PR but include it in the next PR.
Thanks for this excellent pickup.